packages/warehouses/src/warehouseClients/WarehouseBaseClient.ts

```diff
@@ -39,13 +39,40 @@ export default class WarehouseBaseClient<T extends CreateWarehouseCredentials>
         throw new Error('Warehouse method not implemented.');
     }
 
-    async runQuery(sql: string): Promise<{
-        fields: Record<string, { type: DimensionType }>;
-        rows: Record<string, any>[];
-    }> {
+    async streamQuery(
+        query: string,
+        streamCallback: (data: WarehouseResults) => void,
+        options: {
+            tags?: Record<string, string>;
+            timezone?: string;
+        },
+    ): Promise<void> {
         throw new Error('Warehouse method not implemented.');
     }
 
+    async runQuery(
+        sql: string,
+        tags?: Record<string, string>,
+        timezone?: string,
+    ) {
+        let fields: WarehouseResults['fields'] = {};
+        const rows: WarehouseResults['rows'] = [];
```

Comment:
nice wrapping, but I'm wondering if copying the `data.rows` into  a new `rows` means duplicating all the memory usage for each request now

Reply:
It should be the same amount of memory as before as the current code does the same thing. 🤔 
`rows` memory will increase as it gets more entries and `data.rows` only exists while handling that chunk.